lms/static/js/student_account/components/StudentAccountDeletionModal.jsx

```diff
@@ -0,0 +1,212 @@
+/* globals gettext */
+/* eslint-disable react/no-danger */
+import React from 'react';
+import 'whatwg-fetch';
+import PropTypes from 'prop-types';
+import { Button, Modal, Icon, InputText, StatusAlert } from '@edx/paragon/static';
+import StringUtils from 'edx-ui-toolkit/js/utils/string-utils';
+
+class StudentAccountDeletionConfirmationModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.closeConfirmationModal = this.closeConfirmationModal.bind(this);
+    this.deleteAccount = this.deleteAccount.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.passwordFieldValidation = this.passwordFieldValidation.bind(this);
+    this.state = {
+      password: '',
+      passwordSubmitted: false,
+      isValid: true,
+      validationMessage: '',
+      accountQueuedForDeletion: false,
+      responseError: false,
+      responseErrorBody: {},
+    };
+  }
+
+  closeConfirmationModal() {
+    const { onClose } = this.props;
+
+    onClose();
+  }
+
+  deleteAccount() {
+    const { password } = this.state;
+
+    this.setState({ passwordSubmitted: true });
```

Comment:
Can this be moved into the `response.ok` `setState`? 

In the case where `!response.ok`, `failedSubmission` will set `passwordSubmitted` to `false`. I guess this could technically save a render.

Reply:
It's tracking that a request has been submitted, so if it's added within ```response.ok``` it's happening too late. I meant to add logic to not allow multiple submissions until a response has returned. Once I've added that it will be clearer.

Reply:
Actually it is already added, look at the submission button ```disabled``` prop.